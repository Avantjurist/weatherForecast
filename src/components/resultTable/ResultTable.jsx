import React from 'react';
import ForecastItem from './forecastItem/ForecastItem';
import { connect } from 'react-redux'

class ResultTable extends React.Component {
    constructor() {
        super();
        this.state = { forecastItems: [{name: "a"}, {name: "b"}, {name: "c"}]}
    }

    render() {
        console.log("props with cities", this.props);
        return (
            <ul>{this.props.cities.map((forecastItem) => <ForecastItem key={forecastItem.id} item={forecastItem}/>)}</ul>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        cities: state.cities
    }
}

export default connect(mapStateToProps)(ResultTable)