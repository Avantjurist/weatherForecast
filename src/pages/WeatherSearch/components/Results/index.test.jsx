import React from "react";
import { shallow } from "enzyme";
import ResultTable from "./index.component";

describe("Result table", () => {
    it("render correctly with props", () => {
        const props = { forecasts: [{id:1}, {id:2}], fetchData: jest.fn(), match: { params: { city: "test" } } };
        const output = shallow(
            < ResultTable { ...props } />
        )

        expect(output).toMatchSnapshot();
    })
})