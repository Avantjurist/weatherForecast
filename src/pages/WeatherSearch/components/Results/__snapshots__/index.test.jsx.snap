// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result table render correctly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultTable
    fetchData={
      [MockFunction] {
        "calls": Array [
          Array [
            "test",
          ],
        ],
      }
    }
    forecasts={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ]
    }
    match={
      Object {
        "params": Object {
          "city": "test",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForecastItem
          item={
            Object {
              "id": 1,
            }
          }
        />,
        <ForecastItem
          item={
            Object {
              "id": 2,
            }
          }
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "id": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "id": 2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForecastItem
            item={
              Object {
                "id": 1,
              }
            }
          />,
          <ForecastItem
            item={
              Object {
                "id": 2,
              }
            }
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
